src/gui/profilepagewidget.cpp

```diff
@@ -0,0 +1,82 @@
+#include "profilepagewidget.h"
+#include "guiutility.h"
+#include "theme.h"
+
+#include <QPushButton>
+
+namespace OCC {
+
+ProfilePageWidget::ProfilePageWidget(QWidget *parent)
+    : QWidget(parent)
+{
+}
+
+void ProfilePageWidget::setProfileConnector(
+    std::unique_ptr<OcsProfileConnector> profileConnector, const QString &userId)
+{
+    _profileConnector = std::move(profileConnector);
+    _profileConnector->fetchHovercard(userId);
+    connect(
```

Comment:
@FlexW I know `_profileConnector->fetchHovercard(userId);` will not get executed immediately and we can add `signal->slot` connections after this call, but, wouldn't it make more sense to put it as the last line as we do for jobs in rest of ode? For me, it just makes more sense that we connect slots and do the call as the last step.

Reply:
Yes, makes sense.